### PR TITLE
MDX: Better ergonomics for documenting CSF

### PR DESCRIPTION
**Pull request by shilman**
Issue: #8311 

## What I did

- [x] Added imported story function support to MDX
- [x] Improved MDX documentation of CSF

## How to test

See included examples & snapshot tests

### CODE REVIEW

File: addons/docs/docs/recipes.md [hunk 1]
```diff
@@ -31,46 +31,50 @@ The only limitation is that your exported titles (CSF: `default.title`, MDX `Met
...
+**Button.stories.js**
+
+```js
+import React from 'react';
+import { Button } from './Button';
+
+export default {
+  title: 'Demo/Button',
+  component: Button,
+  includeStories: [], // or simply don't load this file at all
```
Comment (tmeasday) [inline addons/docs/docs/recipes.md]: Should we allow `false` as a value here?

File: addons/docs/docs/recipes.md [hunk 2]
```diff
@@ -31,46 +31,50 @@ The only limitation is that your exported titles (CSF: `default.title`, MDX `Met
...
+};
+```
+
+**Button.stories.mdx**
 
 ```md
-import { Story } from '@storybook/addon-docs/blocks';
-import { SomeComponent } from 'somewhere';
+import { Meta, Story } from '@storybook/addon-docs/blocks';
+import \* as stories from './Button.stories.js';
```
Comment (tmeasday) [inline addons/docs/docs/recipes.md]:
```suggestion
import * as stories from './Button.stories.js';
```

File: addons/docs/docs/recipes.md [hunk 3]
```diff
@@ -31,46 +31,50 @@ The only limitation is that your exported titles (CSF: `default.title`, MDX `Met
...
+import { SomeComponent } from 'path/to/SomeComponent';
+
+<Meta {...stories.default} />
```
Comment (tmeasday) [inline addons/docs/docs/recipes.md]: What about the `includeStories`?

File: addons/docs/docs/recipes.md [hunk 4]
```diff
@@ -31,46 +31,50 @@ The only limitation is that your exported titles (CSF: `default.title`, MDX `Met
...
 
 # Button
 
 I can embed a story (but not define one, since this file should not contain a `Meta`):
```
Comment (tmeasday) [inline addons/docs/docs/recipes.md]: This line seems inaccurate now

File: addons/docs/src/mdx/__snapshots__/mdx-compiler-plugin.test.js.snap [hunk 5]
```diff
@@ -51,6 +51,8 @@ function MDXContent({ components, ...props }) {
 
 MDXContent.isMDXComponent = true;
 
+const makeStoryFn = val => (typeof val === 'function' ? val : () => val);
```
Comment (tmeasday) [inline addons/docs/src/mdx/__snapshots__/mdx-compiler-plugin.test.js.snap]: Maybe this should be an import? Or defined inline?

**Review by tmeasday — CHANGES_REQUESTED** [review]
A few minor things

File: addons/docs/docs/recipes.md (hunk 3 above)
Comment (shilman) [reply addons/docs/docs/recipes.md]: It's ignored by the `Meta` block

File: addons/docs/docs/recipes.md (hunk 1 above)
Comment (shilman) [reply addons/docs/docs/recipes.md]: Overloading the type is more trouble than it's worth IMHO

File: addons/docs/src/mdx/__snapshots__/mdx-compiler-plugin.test.js.snap (hunk 5 above)
Comment (shilman) [reply addons/docs/src/mdx/__snapshots__/mdx-compiler-plugin.test.js.snap]: fixed

**Comment by shilman** [comment]
@tmeasday fixed

File: addons/docs/docs/recipes.md (hunk 1 above)
Comment (tmeasday) [reply addons/docs/docs/recipes.md]: OK, that's fair.

File: addons/docs/docs/recipes.md (hunk 3 above)
Comment (tmeasday) [reply addons/docs/docs/recipes.md]: 🤔 that seems a bit odd

**Review by tmeasday — APPROVED** [review]

**Pull request merged**